{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:48.651699Z",
     "start_time": "2022-09-28T04:58:48.648394Z"
    }
   },
   "outputs": [],
   "source": [
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:48.922995Z",
     "start_time": "2022-09-28T04:58:48.654640Z"
    }
   },
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:49.168852Z",
     "start_time": "2022-09-28T04:58:48.926428Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pheweb.json\n",
      "gwascatalog.json\n",
      "pgscatalog.json\n",
      "metal.json\n",
      "regenie.json\n",
      "ldsc.json\n",
      "plink2.json\n",
      "vcf.json\n",
      "fastgwa.json\n",
      "pgscatalog_hm.json\n",
      "plink.json\n",
      "gwaslab.json\n",
      "template.json\n",
      "mrmega.json\n",
      "ssf.json\n",
      "gwascatalog_hm.json\n",
      "saige.json\n"
     ]
    }
   ],
   "source": [
    "alldic={}\n",
    "for i in os.listdir(\"./formats/\"):\n",
    "    if i[-4:]==\"json\":\n",
    "        print(i)\n",
    "        with open(\"./formats/\"+i) as f:\n",
    "            dic=json.load(f)\n",
    "            alldic[i[:-5]]=dic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:49.198503Z",
     "start_time": "2022-09-28T04:58:49.171919Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'pheweb': {'meta_data': {'format_name': 'pheweb',\n",
       "   'format_source': 'https://github.com/statgen/pheweb',\n",
       "   'format_version': '20220928'},\n",
       "  'format_dict': {'rsids': 'rsID',\n",
       "   'chrom': 'CHR',\n",
       "   'pos': 'POS',\n",
       "   'ref': 'NEA',\n",
       "   'alt': 'EA',\n",
       "   'beta': 'BETA',\n",
       "   'af': 'EAF',\n",
       "   'sebeta': 'SE',\n",
       "   'pval': 'P',\n",
       "   'or': 'OR',\n",
       "   'r2': 'INFO',\n",
       "   'num_samples': 'n'}},\n",
       " 'gwascatalog': {'meta_data': {'format_name': 'gwascatalog',\n",
       "   'format_source': 'https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'variant_id': 'SNPID',\n",
       "   'chromosome': 'CHR',\n",
       "   'base_pair_location': 'POS',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'beta': 'BETA',\n",
       "   'effect_allele_frequency': 'EAF',\n",
       "   'standard_error': 'SE',\n",
       "   'p-value': 'P',\n",
       "   'odds_ratio': 'OR',\n",
       "   'ci_lower': 'OR_95L',\n",
       "   'ci_upper': 'OR_95U'}},\n",
       " 'pgscatalog': {'meta_data': {'format_name': 'pgscatalog',\n",
       "   'format_source': 'https://www.pgscatalog.org/downloads/',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'rsID': 'rsID',\n",
       "   'chr_name': 'CHR',\n",
       "   'chr_position': 'POS',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'allelefrequency_effect': 'EAF',\n",
       "   'effect_weight': 'BETA',\n",
       "   'OR': 'OR'}},\n",
       " 'metal': {'meta_data': {'format_name': 'metal',\n",
       "   'format_source': 'https://genome.sph.umich.edu/wiki/METAL_Documentation',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'MarkerName': 'SNPID',\n",
       "   'Allele1': 'EA',\n",
       "   'Allele2': 'NEA',\n",
       "   'Freq1': 'EAF',\n",
       "   'Effect': 'BETA',\n",
       "   'StdErr': 'SE',\n",
       "   'P-value': 'P',\n",
       "   'Direction': 'DIRECTION'}},\n",
       " 'regenie': {'meta_data': {'format_name': 'regenie',\n",
       "   'format_source_1': 'https://rgcgithub.github.io/regenie/',\n",
       "   'format_source_2': 'https://github.com/rgcgithub/regenie/blob/master/example/test_bin_out_firth_Y1.regenie',\n",
       "   'format_version': '20220726',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'ID': 'SNPID',\n",
       "   'CHROM': 'CHR',\n",
       "   'GENPOS': 'POS',\n",
       "   'ALLELE0': 'NEA',\n",
       "   'ALLELE1': 'EA',\n",
       "   'A1FREQ': 'EAF',\n",
       "   'N': 'N',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'CHISQ': 'CHISQ',\n",
       "   'LOG10P': 'MLOG10P'}},\n",
       " 'ldsc': {'meta_data': {'format_name': 'ldsc',\n",
       "   'format_source': 'https://github.com/bulik/ldsc/wiki/Summary-Statistics-File-Format',\n",
       "   'format_source2': 'https://github.com/bulik/ldsc/blob/master/munge_sumstats.py',\n",
       "   'format_version': 20150306},\n",
       "  'format_dict': {'SNP': 'rsID',\n",
       "   'A2': 'NEA',\n",
       "   'A1': 'EA',\n",
       "   'N': 'N',\n",
       "   'Beta': 'BETA',\n",
       "   'P': 'P',\n",
       "   'INFO': 'INFO',\n",
       "   'OR': 'OR'}},\n",
       " 'plink2': {'meta_data': {'format_name': 'PLINK2 .glm.firth, .glm.logistic,.glm.linear',\n",
       "   'format_source': 'https://www.cog-genomics.org/plink/2.0/formats',\n",
       "   'format_version': 'Alpha 3.3 final (3 Jun)',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'ID': 'SNPID',\n",
       "   'CHROM': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'REF': 'NEA',\n",
       "   'ALT': 'EA',\n",
       "   'OBS_CT': 'N',\n",
       "   'A1_FREQ': 'EAF',\n",
       "   'BETA': 'BETA',\n",
       "   '[LOG(OR)_]SE': 'SE',\n",
       "   'Z_[OR_F_]STAT': 'Z',\n",
       "   'P': 'P',\n",
       "   'MACH_R2': 'INFO',\n",
       "   'OR': 'OR',\n",
       "   'L95': 'OR_95L',\n",
       "   'U95': 'OR_95U'}},\n",
       " 'vcf': {'meta_data': {'format_name': 'vcf',\n",
       "   'format_source': 'https://github.com/MRCIEU/gwas-vcf-specification/tree/1.0.0',\n",
       "   'format_version': 20220923,\n",
       "   'format_citation': 'Lyon, M.S., Andrews, S.J., Elsworth, B. et al. The variant call format provides efficient and robust storage of GWAS summary statistics. Genome Biol 22, 32 (2021). https://doi.org/10.1186/s13059-020-02248-0',\n",
       "   'format_fixed_header': '##fileformat=VCFv4.2\\r\\n##FILTER=<ID=PASS,Description=\"All filters passed\">\\r\\n##INFO=<ID=AF,Number=A,Type=Float,Description=\"Allele Frequency\">\\r\\n##FORMAT=<ID=ES,Number=A,Type=Float,Description=\"Effect size estimate relative to the alternative allele\">\\r\\n##FORMAT=<ID=SE,Number=A,Type=Float,Description=\"Standard error of effect size estimate\">\\r\\n##FORMAT=<ID=LP,Number=A,Type=Float,Description=\"-log10 p-value for effect estimate\">\\r\\n##FORMAT=<ID=AF,Number=A,Type=Float,Description=\"Alternate allele frequency in the association study\">\\r\\n##FORMAT=<ID=SS,Number=A,Type=Float,Description=\"Sample size used to estimate genetic effect\">\\r\\n##FORMAT=<ID=EZ,Number=A,Type=Float,Description=\"Z-score provided if it was used to derive the EFFECT and SE fields\">\\r\\n##FORMAT=<ID=SI,Number=A,Type=Float,Description=\"Accuracy score of summary data imputation\">\\r\\n##FORMAT=<ID=NC,Number=A,Type=Float,Description=\"Number of cases used to estimate genetic effect\">\\r\\n##FORMAT=<ID=ID,Number=1,Type=String,Description=\"Study variant identifier\">\\r\\n##META=<ID=TotalVariants,Number=1,Type=Integer,Description=\"Total number of variants in input\">\\r\\n##META=<ID=VariantsNotRead,Number=1,Type=Integer,Description=\"Number of variants that could not be read\">\\r\\n##META=<ID=HarmonisedVariants,Number=1,Type=Integer,Description=\"Total number of harmonised variants\">\\r\\n##META=<ID=VariantsNotHarmonised,Number=1,Type=Integer,Description=\"Total number of variants that could not be harmonised\">\\r\\n##META=<ID=SwitchedAlleles,Number=1,Type=Integer,Description=\"Total number of variants strand switched\">\\r\\n##META=<ID=TotalControls,Number=1,Type=Integer,Description=\"Total number of controls in the association study\">\\r\\n##META=<ID=TotalCases,Number=1,Type=Integer,Description=\"Total number of cases in the association study\">\\r\\n##META=<ID=StudyType,Number=1,Type=String,Description=\"Type of GWAS study [Continuous or CaseControl]\">',\n",
       "   'format_fixed': ['#CHROM',\n",
       "    'POS',\n",
       "    'ID',\n",
       "    'REF',\n",
       "    'ALT',\n",
       "    'QUAL',\n",
       "    'FILTER',\n",
       "    'INFO',\n",
       "    'FORMAT'],\n",
       "   'format_format': ['ID', 'SS', 'ES', 'SE', 'LP', 'SI', 'EZ'],\n",
       "   'format_contig_19': '##contig=<ID=1,length=249250621,assembly=HG19/GRCh37>\\r\\n##contig=<ID=2,length=243199373,assembly=HG19/GRCh37>\\r\\n##contig=<ID=3,length=198022430,assembly=HG19/GRCh37>\\r\\n##contig=<ID=4,length=191154276,assembly=HG19/GRCh37>\\r\\n##contig=<ID=5,length=180915260,assembly=HG19/GRCh37>\\r\\n##contig=<ID=6,length=171115067,assembly=HG19/GRCh37>\\r\\n##contig=<ID=7,length=159138663,assembly=HG19/GRCh37>\\r\\n##contig=<ID=8,length=146364022,assembly=HG19/GRCh37>\\r\\n##contig=<ID=9,length=141213431,assembly=HG19/GRCh37>\\r\\n##contig=<ID=10,length=135534747,assembly=HG19/GRCh37>\\r\\n##contig=<ID=11,length=135006516,assembly=HG19/GRCh37>\\r\\n##contig=<ID=12,length=133851895,assembly=HG19/GRCh37>\\r\\n##contig=<ID=13,length=115169878,assembly=HG19/GRCh37>\\r\\n##contig=<ID=14,length=107349540,assembly=HG19/GRCh37>\\r\\n##contig=<ID=15,length=102531392,assembly=HG19/GRCh37>\\r\\n##contig=<ID=16,length=90354753,assembly=HG19/GRCh37>\\r\\n##contig=<ID=17,length=81195210,assembly=HG19/GRCh37>\\r\\n##contig=<ID=18,length=78077248,assembly=HG19/GRCh37>\\r\\n##contig=<ID=19,length=59128983,assembly=HG19/GRCh37>\\r\\n##contig=<ID=20,length=63025520,assembly=HG19/GRCh37>\\r\\n##contig=<ID=21,length=48129895,assembly=HG19/GRCh37>\\r\\n##contig=<ID=22,length=51304566,assembly=HG19/GRCh37>\\r\\n##contig=<ID=23,length=155270560,assembly=HG19/GRCh37>\\r\\n##contig=<ID=24,length=59373566,assembly=HG19/GRCh37>\\r\\n##contig=<ID=25,length=16569,assembly=HG19/GRCh37>',\n",
       "   'format_contig_38': '##contig=<ID=1,length=248956422,assembly=HG38/GRCh38>\\r\\n##contig=<ID=2,length=242193529,assembly=HG38/GRCh38>\\r\\n##contig=<ID=3,length=198295559,assembly=HG38/GRCh38>\\r\\n##contig=<ID=4,length=190214555,assembly=HG38/GRCh38>\\r\\n##contig=<ID=5,length=181538259,assembly=HG38/GRCh38>\\r\\n##contig=<ID=6,length=170805979,assembly=HG38/GRCh38>\\r\\n##contig=<ID=7,length=159345973,assembly=HG38/GRCh38>\\r\\n##contig=<ID=8,length=145138636,assembly=HG38/GRCh38>\\r\\n##contig=<ID=9,length=138394717,assembly=HG38/GRCh38>\\r\\n##contig=<ID=10,length=133797422,assembly=HG38/GRCh38>\\r\\n##contig=<ID=11,length=135086622,assembly=HG38/GRCh38>\\r\\n##contig=<ID=12,length=133275309,assembly=HG38/GRCh38>\\r\\n##contig=<ID=13,length=114364328,assembly=HG38/GRCh38>\\r\\n##contig=<ID=14,length=107043718,assembly=HG38/GRCh38>\\r\\n##contig=<ID=15,length=101991189,assembly=HG38/GRCh38>\\r\\n##contig=<ID=16,length=90338345,assembly=HG38/GRCh38>\\r\\n##contig=<ID=17,length=83257441,assembly=HG38/GRCh38>\\r\\n##contig=<ID=18,length=80373285,assembly=HG38/GRCh38>\\r\\n##contig=<ID=19,length=58617616,assembly=HG38/GRCh38>\\r\\n##contig=<ID=20,length=64444167,assembly=HG38/GRCh38>\\r\\n##contig=<IDs=21,length=46709983,assembly=HG38/GRCh38>\\r\\n##contig=<ID=22,length=50818468,assembly=HG38/GRCh38>\\r\\n##contig=<ID=23,length=156040895,assembly=HG38/GRCh38>\\r\\n##contig=<ID=24,length=57227415,assembly=HG38/GRCh38>\\r\\n##contig=<ID=25,length=16569,assembly=HG38/GRCh38>'},\n",
       "  'format_dict': {'ID': 'rsID',\n",
       "   '#CHROM': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'REF': 'NEA',\n",
       "   'ALT': 'EA',\n",
       "   'SS': 'N',\n",
       "   'AF': 'EAF',\n",
       "   'ES': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'LP': 'MLOG10P',\n",
       "   'SI': 'INFO',\n",
       "   'EZ': 'Z'}},\n",
       " 'fastgwa': {'meta_data': {'format_name': 'fastgwa',\n",
       "   'format_source': 'https://yanglab.westlake.edu.cn/software/gcta/#fastGWA',\n",
       "   'format_source2': 'https://yanglab.westlake.edu.cn/software/gcta/#fastGWA-GLMM',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'SNP': 'SNPID',\n",
       "   'CHR': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'A2': 'NEA',\n",
       "   'A1': 'EA',\n",
       "   'AF1': 'EAF',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'P': 'P'}},\n",
       " 'pgscatalog_hm': {'meta_data': {'format_name': 'pgscatalog_hm Harmonized File',\n",
       "   'format_source': 'https://www.pgscatalog.org/downloads/',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'hm_rsID': 'rsID',\n",
       "   'hm_chr': 'CHR',\n",
       "   'hm_pos': 'POS',\n",
       "   'hm_inferOtherAllele': 'NEA',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'allelefrequency_effect': 'EAF',\n",
       "   'effect_weight': 'BETA',\n",
       "   'OR': 'OR'}},\n",
       " 'plink': {'meta_data': {'format_name': '.assoc.linear, .assoc.logistic, .assoc.dosage',\n",
       "   'format_source': 'https://www.cog-genomics.org/plink/1.9/',\n",
       "   'format_version': 'Stable (beta 6.26, 2 Apr)'},\n",
       "  'format_dict': {'SNP': 'rsID',\n",
       "   'CHR': 'CHR',\n",
       "   'BP': 'POS',\n",
       "   'A2': 'NEA',\n",
       "   'A1': 'EA',\n",
       "   'BETA': 'BETA',\n",
       "   'OR': 'OR',\n",
       "   'SE': 'SE',\n",
       "   'P': 'P',\n",
       "   'INFO': 'INFO',\n",
       "   'L95': 'OR_95L',\n",
       "   'H95': 'OR_95U',\n",
       "   'STAT': 'Z',\n",
       "   'NMISS': 'N'}},\n",
       " 'gwaslab': {'meta_data': {'format_name': 'gwaslab',\n",
       "   'format_source': 'https://cloufield.github.io/gwaslab/',\n",
       "   'format_version': '20220729_v3'},\n",
       "  'format_dict': {'SNPID': 'SNPID',\n",
       "   'rsID': 'rsID',\n",
       "   'CHR': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'NEA': 'NEA',\n",
       "   'EA': 'EA',\n",
       "   'N': 'N',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'P': 'P',\n",
       "   'INFO': 'INFO',\n",
       "   'OR': 'OR',\n",
       "   'OR_95L': 'OR_95L',\n",
       "   'OR_95U': 'OR_95U',\n",
       "   'MLOG10': 'MLOG10',\n",
       "   'CHISQ': 'CHISQ',\n",
       "   'Z': 'Z',\n",
       "   'DIRECTION': 'DIRECTION',\n",
       "   'STATUS': 'STATUS'}},\n",
       " 'template': {'meta_data': {'format_name': '',\n",
       "   'format_source': '',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'': 'OR_95U'}},\n",
       " 'mrmega': {'meta_data': {'format_name': 'mrmega',\n",
       "   'format_source': 'https://genomics.ut.ee/en/tools',\n",
       "   'format_version': 'ver0.2',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'MarkerName ': 'SNPID',\n",
       "   'Chromosome': 'CHR',\n",
       "   'Position': 'POS',\n",
       "   'NEA': 'NEA',\n",
       "   'EA': 'EA',\n",
       "   'EAF': 'EAF',\n",
       "   'Nsample': 'N',\n",
       "   'Effects': 'DIRECTION',\n",
       "   'chisq_association': 'CHISQ',\n",
       "   'ndf_association': 'ndf_association',\n",
       "   'P-value_association': 'P'}},\n",
       " 'ssf': {'meta_data': {'format_name': 'ssf',\n",
       "   'format_source': 'https://www.biorxiv.org/content/10.1101/2022.07.15.500230v1.full',\n",
       "   'format_version': 20220726},\n",
       "  'format_dict': {'variant_id': 'SNPID',\n",
       "   'rsid': 'rsID',\n",
       "   'chromosome': 'CHR',\n",
       "   'bas_pair_location': 'POS',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'n': 'N',\n",
       "   'beta': 'BETA',\n",
       "   'standard_error': 'SE',\n",
       "   'p_value': 'P',\n",
       "   'info': 'INFO',\n",
       "   'odds_ratio': 'OR',\n",
       "   'ci_lower': 'OR_95L',\n",
       "   'ci_upper': 'OR_95U'}},\n",
       " 'gwascatalog_hm': {'meta_data': {'format_name': 'gwascatalog_hm',\n",
       "   'format_source': 'https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics',\n",
       "   'format_version': 20220806},\n",
       "  'format_dict': {'hm_variant_id': 'SNPID',\n",
       "   'chromosome': 'CHR',\n",
       "   'base_pair_location': 'POS',\n",
       "   'hm_other_allele': 'NEA',\n",
       "   'hm_effect_allele': 'EA',\n",
       "   'hm_beta': 'BETA',\n",
       "   'hm_effect_allele_frequency': 'EAF',\n",
       "   'standard_error': 'SE',\n",
       "   'p-value': 'P',\n",
       "   'hm_odds_ratio': 'OR',\n",
       "   'hm_ci_lower': 'OR_95L',\n",
       "   'hm_ci_upper': 'OR_95U'}},\n",
       " 'saige': {'meta_data': {'format_name': 'saige',\n",
       "   'format_source': 'https://saigegit.github.io//SAIGE-doc/docs/single_step2.html',\n",
       "   'format_version': 'v1.1.3',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'SNPID': 'SNPID',\n",
       "   'CHR': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'Allele1': 'NEA',\n",
       "   'Allele2': 'EA',\n",
       "   'AF_Allele2': 'EAF',\n",
       "   'N': 'N',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'p.value': 'P',\n",
       "   'imputationInfo': 'INFO'}}}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alldic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:49.233198Z",
     "start_time": "2022-09-28T04:58:49.200611Z"
    }
   },
   "outputs": [],
   "source": [
    "with open('./formatbook.json', 'w') as f:\n",
    "    json.dump(alldic,f,indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:49.304249Z",
     "start_time": "2022-09-28T04:58:49.235880Z"
    }
   },
   "outputs": [],
   "source": [
    "book=json.load(open(\"formatbook.json\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-28T04:58:49.326855Z",
     "start_time": "2022-09-28T04:58:49.306944Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'pheweb': {'meta_data': {'format_name': 'pheweb',\n",
       "   'format_source': 'https://github.com/statgen/pheweb',\n",
       "   'format_version': '20220928'},\n",
       "  'format_dict': {'rsids': 'rsID',\n",
       "   'chrom': 'CHR',\n",
       "   'pos': 'POS',\n",
       "   'ref': 'NEA',\n",
       "   'alt': 'EA',\n",
       "   'beta': 'BETA',\n",
       "   'af': 'EAF',\n",
       "   'sebeta': 'SE',\n",
       "   'pval': 'P',\n",
       "   'or': 'OR',\n",
       "   'r2': 'INFO',\n",
       "   'num_samples': 'n'}},\n",
       " 'gwascatalog': {'meta_data': {'format_name': 'gwascatalog',\n",
       "   'format_source': 'https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'variant_id': 'SNPID',\n",
       "   'chromosome': 'CHR',\n",
       "   'base_pair_location': 'POS',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'beta': 'BETA',\n",
       "   'effect_allele_frequency': 'EAF',\n",
       "   'standard_error': 'SE',\n",
       "   'p-value': 'P',\n",
       "   'odds_ratio': 'OR',\n",
       "   'ci_lower': 'OR_95L',\n",
       "   'ci_upper': 'OR_95U'}},\n",
       " 'pgscatalog': {'meta_data': {'format_name': 'pgscatalog',\n",
       "   'format_source': 'https://www.pgscatalog.org/downloads/',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'rsID': 'rsID',\n",
       "   'chr_name': 'CHR',\n",
       "   'chr_position': 'POS',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'allelefrequency_effect': 'EAF',\n",
       "   'effect_weight': 'BETA',\n",
       "   'OR': 'OR'}},\n",
       " 'metal': {'meta_data': {'format_name': 'metal',\n",
       "   'format_source': 'https://genome.sph.umich.edu/wiki/METAL_Documentation',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'MarkerName': 'SNPID',\n",
       "   'Allele1': 'EA',\n",
       "   'Allele2': 'NEA',\n",
       "   'Freq1': 'EAF',\n",
       "   'Effect': 'BETA',\n",
       "   'StdErr': 'SE',\n",
       "   'P-value': 'P',\n",
       "   'Direction': 'DIRECTION'}},\n",
       " 'regenie': {'meta_data': {'format_name': 'regenie',\n",
       "   'format_source_1': 'https://rgcgithub.github.io/regenie/',\n",
       "   'format_source_2': 'https://github.com/rgcgithub/regenie/blob/master/example/test_bin_out_firth_Y1.regenie',\n",
       "   'format_version': '20220726',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'ID': 'SNPID',\n",
       "   'CHROM': 'CHR',\n",
       "   'GENPOS': 'POS',\n",
       "   'ALLELE0': 'NEA',\n",
       "   'ALLELE1': 'EA',\n",
       "   'A1FREQ': 'EAF',\n",
       "   'N': 'N',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'CHISQ': 'CHISQ',\n",
       "   'LOG10P': 'MLOG10P'}},\n",
       " 'ldsc': {'meta_data': {'format_name': 'ldsc',\n",
       "   'format_source': 'https://github.com/bulik/ldsc/wiki/Summary-Statistics-File-Format',\n",
       "   'format_source2': 'https://github.com/bulik/ldsc/blob/master/munge_sumstats.py',\n",
       "   'format_version': 20150306},\n",
       "  'format_dict': {'SNP': 'rsID',\n",
       "   'A2': 'NEA',\n",
       "   'A1': 'EA',\n",
       "   'N': 'N',\n",
       "   'Beta': 'BETA',\n",
       "   'P': 'P',\n",
       "   'INFO': 'INFO',\n",
       "   'OR': 'OR'}},\n",
       " 'plink2': {'meta_data': {'format_name': 'PLINK2 .glm.firth, .glm.logistic,.glm.linear',\n",
       "   'format_source': 'https://www.cog-genomics.org/plink/2.0/formats',\n",
       "   'format_version': 'Alpha 3.3 final (3 Jun)',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'ID': 'SNPID',\n",
       "   'CHROM': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'REF': 'NEA',\n",
       "   'ALT': 'EA',\n",
       "   'OBS_CT': 'N',\n",
       "   'A1_FREQ': 'EAF',\n",
       "   'BETA': 'BETA',\n",
       "   '[LOG(OR)_]SE': 'SE',\n",
       "   'Z_[OR_F_]STAT': 'Z',\n",
       "   'P': 'P',\n",
       "   'MACH_R2': 'INFO',\n",
       "   'OR': 'OR',\n",
       "   'L95': 'OR_95L',\n",
       "   'U95': 'OR_95U'}},\n",
       " 'vcf': {'meta_data': {'format_name': 'vcf',\n",
       "   'format_source': 'https://github.com/MRCIEU/gwas-vcf-specification/tree/1.0.0',\n",
       "   'format_version': 20220923,\n",
       "   'format_citation': 'Lyon, M.S., Andrews, S.J., Elsworth, B. et al. The variant call format provides efficient and robust storage of GWAS summary statistics. Genome Biol 22, 32 (2021). https://doi.org/10.1186/s13059-020-02248-0',\n",
       "   'format_fixed_header': '##fileformat=VCFv4.2\\r\\n##FILTER=<ID=PASS,Description=\"All filters passed\">\\r\\n##INFO=<ID=AF,Number=A,Type=Float,Description=\"Allele Frequency\">\\r\\n##FORMAT=<ID=ES,Number=A,Type=Float,Description=\"Effect size estimate relative to the alternative allele\">\\r\\n##FORMAT=<ID=SE,Number=A,Type=Float,Description=\"Standard error of effect size estimate\">\\r\\n##FORMAT=<ID=LP,Number=A,Type=Float,Description=\"-log10 p-value for effect estimate\">\\r\\n##FORMAT=<ID=AF,Number=A,Type=Float,Description=\"Alternate allele frequency in the association study\">\\r\\n##FORMAT=<ID=SS,Number=A,Type=Float,Description=\"Sample size used to estimate genetic effect\">\\r\\n##FORMAT=<ID=EZ,Number=A,Type=Float,Description=\"Z-score provided if it was used to derive the EFFECT and SE fields\">\\r\\n##FORMAT=<ID=SI,Number=A,Type=Float,Description=\"Accuracy score of summary data imputation\">\\r\\n##FORMAT=<ID=NC,Number=A,Type=Float,Description=\"Number of cases used to estimate genetic effect\">\\r\\n##FORMAT=<ID=ID,Number=1,Type=String,Description=\"Study variant identifier\">\\r\\n##META=<ID=TotalVariants,Number=1,Type=Integer,Description=\"Total number of variants in input\">\\r\\n##META=<ID=VariantsNotRead,Number=1,Type=Integer,Description=\"Number of variants that could not be read\">\\r\\n##META=<ID=HarmonisedVariants,Number=1,Type=Integer,Description=\"Total number of harmonised variants\">\\r\\n##META=<ID=VariantsNotHarmonised,Number=1,Type=Integer,Description=\"Total number of variants that could not be harmonised\">\\r\\n##META=<ID=SwitchedAlleles,Number=1,Type=Integer,Description=\"Total number of variants strand switched\">\\r\\n##META=<ID=TotalControls,Number=1,Type=Integer,Description=\"Total number of controls in the association study\">\\r\\n##META=<ID=TotalCases,Number=1,Type=Integer,Description=\"Total number of cases in the association study\">\\r\\n##META=<ID=StudyType,Number=1,Type=String,Description=\"Type of GWAS study [Continuous or CaseControl]\">',\n",
       "   'format_fixed': ['#CHROM',\n",
       "    'POS',\n",
       "    'ID',\n",
       "    'REF',\n",
       "    'ALT',\n",
       "    'QUAL',\n",
       "    'FILTER',\n",
       "    'INFO',\n",
       "    'FORMAT'],\n",
       "   'format_format': ['ID', 'SS', 'ES', 'SE', 'LP', 'SI', 'EZ'],\n",
       "   'format_contig_19': '##contig=<ID=1,length=249250621,assembly=HG19/GRCh37>\\r\\n##contig=<ID=2,length=243199373,assembly=HG19/GRCh37>\\r\\n##contig=<ID=3,length=198022430,assembly=HG19/GRCh37>\\r\\n##contig=<ID=4,length=191154276,assembly=HG19/GRCh37>\\r\\n##contig=<ID=5,length=180915260,assembly=HG19/GRCh37>\\r\\n##contig=<ID=6,length=171115067,assembly=HG19/GRCh37>\\r\\n##contig=<ID=7,length=159138663,assembly=HG19/GRCh37>\\r\\n##contig=<ID=8,length=146364022,assembly=HG19/GRCh37>\\r\\n##contig=<ID=9,length=141213431,assembly=HG19/GRCh37>\\r\\n##contig=<ID=10,length=135534747,assembly=HG19/GRCh37>\\r\\n##contig=<ID=11,length=135006516,assembly=HG19/GRCh37>\\r\\n##contig=<ID=12,length=133851895,assembly=HG19/GRCh37>\\r\\n##contig=<ID=13,length=115169878,assembly=HG19/GRCh37>\\r\\n##contig=<ID=14,length=107349540,assembly=HG19/GRCh37>\\r\\n##contig=<ID=15,length=102531392,assembly=HG19/GRCh37>\\r\\n##contig=<ID=16,length=90354753,assembly=HG19/GRCh37>\\r\\n##contig=<ID=17,length=81195210,assembly=HG19/GRCh37>\\r\\n##contig=<ID=18,length=78077248,assembly=HG19/GRCh37>\\r\\n##contig=<ID=19,length=59128983,assembly=HG19/GRCh37>\\r\\n##contig=<ID=20,length=63025520,assembly=HG19/GRCh37>\\r\\n##contig=<ID=21,length=48129895,assembly=HG19/GRCh37>\\r\\n##contig=<ID=22,length=51304566,assembly=HG19/GRCh37>\\r\\n##contig=<ID=23,length=155270560,assembly=HG19/GRCh37>\\r\\n##contig=<ID=24,length=59373566,assembly=HG19/GRCh37>\\r\\n##contig=<ID=25,length=16569,assembly=HG19/GRCh37>',\n",
       "   'format_contig_38': '##contig=<ID=1,length=248956422,assembly=HG38/GRCh38>\\r\\n##contig=<ID=2,length=242193529,assembly=HG38/GRCh38>\\r\\n##contig=<ID=3,length=198295559,assembly=HG38/GRCh38>\\r\\n##contig=<ID=4,length=190214555,assembly=HG38/GRCh38>\\r\\n##contig=<ID=5,length=181538259,assembly=HG38/GRCh38>\\r\\n##contig=<ID=6,length=170805979,assembly=HG38/GRCh38>\\r\\n##contig=<ID=7,length=159345973,assembly=HG38/GRCh38>\\r\\n##contig=<ID=8,length=145138636,assembly=HG38/GRCh38>\\r\\n##contig=<ID=9,length=138394717,assembly=HG38/GRCh38>\\r\\n##contig=<ID=10,length=133797422,assembly=HG38/GRCh38>\\r\\n##contig=<ID=11,length=135086622,assembly=HG38/GRCh38>\\r\\n##contig=<ID=12,length=133275309,assembly=HG38/GRCh38>\\r\\n##contig=<ID=13,length=114364328,assembly=HG38/GRCh38>\\r\\n##contig=<ID=14,length=107043718,assembly=HG38/GRCh38>\\r\\n##contig=<ID=15,length=101991189,assembly=HG38/GRCh38>\\r\\n##contig=<ID=16,length=90338345,assembly=HG38/GRCh38>\\r\\n##contig=<ID=17,length=83257441,assembly=HG38/GRCh38>\\r\\n##contig=<ID=18,length=80373285,assembly=HG38/GRCh38>\\r\\n##contig=<ID=19,length=58617616,assembly=HG38/GRCh38>\\r\\n##contig=<ID=20,length=64444167,assembly=HG38/GRCh38>\\r\\n##contig=<IDs=21,length=46709983,assembly=HG38/GRCh38>\\r\\n##contig=<ID=22,length=50818468,assembly=HG38/GRCh38>\\r\\n##contig=<ID=23,length=156040895,assembly=HG38/GRCh38>\\r\\n##contig=<ID=24,length=57227415,assembly=HG38/GRCh38>\\r\\n##contig=<ID=25,length=16569,assembly=HG38/GRCh38>'},\n",
       "  'format_dict': {'ID': 'rsID',\n",
       "   '#CHROM': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'REF': 'NEA',\n",
       "   'ALT': 'EA',\n",
       "   'SS': 'N',\n",
       "   'AF': 'EAF',\n",
       "   'ES': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'LP': 'MLOG10P',\n",
       "   'SI': 'INFO',\n",
       "   'EZ': 'Z'}},\n",
       " 'fastgwa': {'meta_data': {'format_name': 'fastgwa',\n",
       "   'format_source': 'https://yanglab.westlake.edu.cn/software/gcta/#fastGWA',\n",
       "   'format_source2': 'https://yanglab.westlake.edu.cn/software/gcta/#fastGWA-GLMM',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'SNP': 'SNPID',\n",
       "   'CHR': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'A2': 'NEA',\n",
       "   'A1': 'EA',\n",
       "   'AF1': 'EAF',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'P': 'P'}},\n",
       " 'pgscatalog_hm': {'meta_data': {'format_name': 'pgscatalog_hm Harmonized File',\n",
       "   'format_source': 'https://www.pgscatalog.org/downloads/',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'hm_rsID': 'rsID',\n",
       "   'hm_chr': 'CHR',\n",
       "   'hm_pos': 'POS',\n",
       "   'hm_inferOtherAllele': 'NEA',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'allelefrequency_effect': 'EAF',\n",
       "   'effect_weight': 'BETA',\n",
       "   'OR': 'OR'}},\n",
       " 'plink': {'meta_data': {'format_name': '.assoc.linear, .assoc.logistic, .assoc.dosage',\n",
       "   'format_source': 'https://www.cog-genomics.org/plink/1.9/',\n",
       "   'format_version': 'Stable (beta 6.26, 2 Apr)'},\n",
       "  'format_dict': {'SNP': 'rsID',\n",
       "   'CHR': 'CHR',\n",
       "   'BP': 'POS',\n",
       "   'A2': 'NEA',\n",
       "   'A1': 'EA',\n",
       "   'BETA': 'BETA',\n",
       "   'OR': 'OR',\n",
       "   'SE': 'SE',\n",
       "   'P': 'P',\n",
       "   'INFO': 'INFO',\n",
       "   'L95': 'OR_95L',\n",
       "   'H95': 'OR_95U',\n",
       "   'STAT': 'Z',\n",
       "   'NMISS': 'N'}},\n",
       " 'gwaslab': {'meta_data': {'format_name': 'gwaslab',\n",
       "   'format_source': 'https://cloufield.github.io/gwaslab/',\n",
       "   'format_version': '20220729_v3'},\n",
       "  'format_dict': {'SNPID': 'SNPID',\n",
       "   'rsID': 'rsID',\n",
       "   'CHR': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'NEA': 'NEA',\n",
       "   'EA': 'EA',\n",
       "   'N': 'N',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'P': 'P',\n",
       "   'INFO': 'INFO',\n",
       "   'OR': 'OR',\n",
       "   'OR_95L': 'OR_95L',\n",
       "   'OR_95U': 'OR_95U',\n",
       "   'MLOG10': 'MLOG10',\n",
       "   'CHISQ': 'CHISQ',\n",
       "   'Z': 'Z',\n",
       "   'DIRECTION': 'DIRECTION',\n",
       "   'STATUS': 'STATUS'}},\n",
       " 'template': {'meta_data': {'format_name': '',\n",
       "   'format_source': '',\n",
       "   'format_version': '20220726'},\n",
       "  'format_dict': {'': 'OR_95U'}},\n",
       " 'mrmega': {'meta_data': {'format_name': 'mrmega',\n",
       "   'format_source': 'https://genomics.ut.ee/en/tools',\n",
       "   'format_version': 'ver0.2',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'MarkerName ': 'SNPID',\n",
       "   'Chromosome': 'CHR',\n",
       "   'Position': 'POS',\n",
       "   'NEA': 'NEA',\n",
       "   'EA': 'EA',\n",
       "   'EAF': 'EAF',\n",
       "   'Nsample': 'N',\n",
       "   'Effects': 'DIRECTION',\n",
       "   'chisq_association': 'CHISQ',\n",
       "   'ndf_association': 'ndf_association',\n",
       "   'P-value_association': 'P'}},\n",
       " 'ssf': {'meta_data': {'format_name': 'ssf',\n",
       "   'format_source': 'https://www.biorxiv.org/content/10.1101/2022.07.15.500230v1.full',\n",
       "   'format_version': 20220726},\n",
       "  'format_dict': {'variant_id': 'SNPID',\n",
       "   'rsid': 'rsID',\n",
       "   'chromosome': 'CHR',\n",
       "   'bas_pair_location': 'POS',\n",
       "   'other_allele': 'NEA',\n",
       "   'effect_allele': 'EA',\n",
       "   'n': 'N',\n",
       "   'beta': 'BETA',\n",
       "   'standard_error': 'SE',\n",
       "   'p_value': 'P',\n",
       "   'info': 'INFO',\n",
       "   'odds_ratio': 'OR',\n",
       "   'ci_lower': 'OR_95L',\n",
       "   'ci_upper': 'OR_95U'}},\n",
       " 'gwascatalog_hm': {'meta_data': {'format_name': 'gwascatalog_hm',\n",
       "   'format_source': 'https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics',\n",
       "   'format_version': 20220806},\n",
       "  'format_dict': {'hm_variant_id': 'SNPID',\n",
       "   'chromosome': 'CHR',\n",
       "   'base_pair_location': 'POS',\n",
       "   'hm_other_allele': 'NEA',\n",
       "   'hm_effect_allele': 'EA',\n",
       "   'hm_beta': 'BETA',\n",
       "   'hm_effect_allele_frequency': 'EAF',\n",
       "   'standard_error': 'SE',\n",
       "   'p-value': 'P',\n",
       "   'hm_odds_ratio': 'OR',\n",
       "   'hm_ci_lower': 'OR_95L',\n",
       "   'hm_ci_upper': 'OR_95U'}},\n",
       " 'saige': {'meta_data': {'format_name': 'saige',\n",
       "   'format_source': 'https://saigegit.github.io//SAIGE-doc/docs/single_step2.html',\n",
       "   'format_version': 'v1.1.3',\n",
       "   'last_check_date': 20220806},\n",
       "  'format_dict': {'SNPID': 'SNPID',\n",
       "   'CHR': 'CHR',\n",
       "   'POS': 'POS',\n",
       "   'Allele1': 'NEA',\n",
       "   'Allele2': 'EA',\n",
       "   'AF_Allele2': 'EAF',\n",
       "   'N': 'N',\n",
       "   'BETA': 'BETA',\n",
       "   'SE': 'SE',\n",
       "   'p.value': 'P',\n",
       "   'imputationInfo': 'INFO'}}}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "book"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
